gadgets/trace_gc/gadget.yaml

```diff
@@ -0,0 +1,64 @@
+name: trace gc
+description: use uprobe to trace garbage collection in Java and Python
```

Comment:
I can split this into two separate gadgets, or adding `py_` and `java_` according to the next comment. Which way do you like better?

Reply:
Actually I think for the image-based gadget, it lacks the ability to choose which tracepoints should be attached.
For example, in builtin-gadgets and bcc, we can write:
```
if args.trace_java {
  attach_java_tracepoints();
}
if args.trace_py {
  attach_py_tracepoints();
}
```
But in image-based gadgets, we have no choice but have to attach to all tracepoints. This is also un-flexible in the `LSM strace` gadget. In built-in gadgets, these are user-specific logics, but we don't have a way to write something like this in image-based gadgets.

So could we design an WASM api for this? Such as calling a wasm function called `init_gadget` (if existing), and it will return which tracepoints to be attached, maybe also which columns to be hidden.

So could we

Reply:
I don't have a strong opinion. I would suggest you do as it is easier to implement.

I have the same dilemna in https://github.com/alban/fsnotify-enricher where users can monitor either fanotify or inotify. I don't think users would want to do both. I chose to keep one gadget because there is a some shared ebpf code for each, so it is easier to maintain. But since I cannot disable an attachment if the user uses `--fanotify-only`, I just return early in the ebpf programs that are not useful. I can revisit this choice if ig gets this WASM API.